src/components/modals/login/CustomConnectButton.tsx

```diff
@@ -0,0 +1,98 @@
+import Button, { ButtonProps } from '@/components/Button'
+import { useMyAccount } from '@/stores/my-account'
+import { ConnectButton } from '@rainbow-me/rainbowkit'
+import { useAccount } from 'wagmi'
+
+type CustomConnectButtonProps = ButtonProps & {
+  className?: string
+  label?: React.ReactNode
+  signAndLinkOnConnect?: boolean
+  signAndLinkEvmAddress: (
+    emvAddress?: string,
+    substrateAddress?: string | null
+  ) => Promise<void>
+  isLoading: boolean
+}
+
+export const CustomConnectButton = ({
+  className,
+  signAndLinkEvmAddress,
+  label = 'Connect EVM Wallet',
+  isLoading,
+  signAndLinkOnConnect = true,
+  ...buttonProps
+}: CustomConnectButtonProps) => {
+  const mySubstrateAddress = useMyAccount((state) => state.address)
+  const { isConnected } = useAccount({
+    onConnect: async ({ address }) => {
+      !isConnected &&
+        signAndLinkOnConnect &&
+        signAndLinkEvmAddress(address, mySubstrateAddress)
+    },
+  })
+
+  return (
+    <ConnectButton.Custom>
+      {({
+        account,
+        chain,
+        openChainModal,
+        openConnectModal,
+        authenticationStatus,
+        mounted,
+      }) => {
+        const ready = mounted && authenticationStatus !== 'loading'
+        const connected =
+          ready &&
+          account &&
+          chain &&
+          (!authenticationStatus || authenticationStatus === 'authenticated')
+
+        if (!connected) {
+          return (
+            <Button
+              onClick={openConnectModal}
+              size={'lg'}
+              isLoading={isLoading}
+              className={className}
+              disabled={isLoading}
+              {...buttonProps}
```

Comment:
I think you can extract common props for all the buttons below too to a new variable (isLoading, isDisabled, className, etc.)

and actually the disabled one you can remove it, because if it is loading, its also automatically disabled.